jre_emul/android/platform/libcore/ojluni/src/main/java/java/io/File.java

```diff
@@ -1281,4 +1284,73 @@ private static native boolean isCaseSensitiveImpl() /*-[
       }
       return true;
     ]-*/;
+
+    // -- Integration with java.nio.file --
+
+    private volatile transient Path filePath;
+
+    /**
+     * Returns a {@link Path java.nio.file.Path} object constructed from the
+     * this abstract path. The resulting {@code Path} is associated with the
+     * {@link java.nio.file.FileSystems#getDefault default-filesystem}.
+     *
+     * <p> The first invocation of this method works as if invoking it were
+     * equivalent to evaluating the expression:
+     * <blockquote><pre>
+     * {@link java.nio.file.FileSystems#getDefault FileSystems.getDefault}().{@link
+     * java.nio.file.FileSystem#getPath getPath}(this.{@link #getPath getPath}());
+     * </pre></blockquote>
+     * Subsequent invocations of this method return the same {@code Path}.
+     *
+     * <p> If this abstract pathname is the empty abstract pathname then this
+     * method returns a {@code Path} that may be used to access the current
+     * user directory.
+     *
+     * @return  a {@code Path} constructed from this abstract path
+     *
+     * @throws  java.nio.file.InvalidPathException
+     *          if a {@code Path} object cannot be constructed from the abstract
+     *          path (see {@link java.nio.file.FileSystem#getPath FileSystem.getPath})
+     *
+     * @since   1.7
+     * @see Path#toFile
+     */
+    public Path toPath() {
+        Path result = filePath;
+        if (result == null) {
+            synchronized (this) {
+                result = filePath;
+                if (result == null) {
+                    try {
+                        Method getDefaultMethod;
+                        try {
+                            getDefaultMethod = Class.forName("java.nio.file.FileSystems")
+                                    .getDeclaredMethod("getDefault");  // no parameters
+                        } catch (ClassNotFoundException e) {
+                            throw new LibraryNotLinkedError("Get default FileSystem",
+                                    "jre_file", "java.nio.file.FileSystems");
```

Comment:
The first parameter should say what library subset missing, so "java.nio.file support" is more general. 

The last parameter needs to be camel-cased (so "JavaNioFileFileSystems"). This is because that's the missing symbol name which gets reported as a linker error.